app/controllers/users_controller.rb

```diff
@@ -0,0 +1,64 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    find_user
+  end
+
+  def login_form
+  end
+
+  def login
+    if session[:user_id] == nil
+      user = User.find_by(name: params[:name])
+      if user
+        session[:user_id] = user.id
+        flash[:message] = "Successfully logged in as existing user #{ user.name }"
+        redirect_to root_path
+      else
+        create
+      end
+    else
+      flash[:status] = :failure
+      flash[:message] = "You are already logged in as #{User.find_by(id: session[:user_id].to_i).name}"
+      redirect_to root_path
+    end
+  end
+
+  def create
+    unless params {name.blank?}
+      if User.create(name: params[:name])
+        user = User.find_by(name: params[:name])
```

Comment:
The name of this method confused me at first - `create` is usually a controller action, but here you have it as a helper method. I do like that you've extracted this functionality, but it might be wise to rename it to something like `build_new_user` and mark it as `private`.